src/widgets/refinement-list/refinement-list.tsx

```diff
@@ -413,15 +419,21 @@ const refinementList: RefinementListWidget = function refinementList(
       ),
     },
   };
+  const templates = {
+    ...defaultTemplates,
+    ...userTemplates,
+  };
 
   const specializedRenderer = renderer({
     containerNode,
     cssClasses,
-    templates: userTemplates,
+    templates,
     searchBoxTemplates: {
-      submit: userTemplates.searchableSubmit,
-      reset: userTemplates.searchableReset,
-      loadingIndicator: userTemplates.searchableLoadingIndicator,
+      submit: templates.searchableSubmit || searchBoxDefaultTemplates.submit,
+      reset: templates.searchableReset || searchBoxDefaultTemplates.reset,
+      loadingIndicator:
+        templates.searchableLoadingIndicator ||
+        searchBoxDefaultTemplates.loadingIndicator,
```

Comment:
Why did this logic have to change? IIUC it slightly changes the behavior.

Reply:
If I've understood correctly how it behaves, the logic changed because the renderer is supposed to receive all the templates. It was previously not throwing an error since we partially accepted templates, but we were spreading the `searchBoxDefaultTemplates` anyway.

Thinking again, since we spread the `searchBoxDefaultTemplates` anyway, it probably make sense to make it partial in the `renderer`, wdyt?

Reply:
agreed, since we spread in the template, we should accept partial searchbox templates